### DISCUSSION TITLE
wasm-mutate

### DISCUSSION THREAD

**Discussion by pgarba** [bytecodealliance/wasm-tools#1953] [Q&A]
Hi,

I'm playing around with wasm-mutate and noticed that after some iterations it will break the binary.

Error message when loading the wasm file in Firefox:
`"wasm streaming compile failed: CompileError: wasm validation error: at offset 398: exported table index out of bounds"`

Sometimes it just breaks the code and the returned value is wrong.

Is there any stable version of the tool ? Right now I'm compiling from the master branch.

**Comment by alexcrichton** [comment]
It's a bug if wasm-mutate creates an invalid module from a valid module. Would you be able to share a test case?
